### DISCUSSION TITLE
What happens to a Persistent Volume when a node is removed in Kubernetes?

### DISCUSSION THREAD

**Discussion by toanalien** [vitobotta/hetzner-k3s#462] [Q&A] [answered]
If I remove a node from the cluster (already removed in Kubernetes) and the volume (PV) was previously attached to node1, what happens to the volume when node1 is deleted? Do I need to manually reattach the volume to another node, or will it be automatically handled?

**Comment by vitobotta** [comment, marked as answer]
Converting into a discussion since this is not about a problem with the tool. In the future please use issues with problems with hetzner-k3s, and discussion for questions etc. :)

When the pod using a volume is evicted, the CSI controller detaches the volume from that node, and reattaches it to whatever node the pod is scheduled on afterwards. You don't need to do anything manually, it's all handled automatically by the CSI.

**Reply by toanalien** [reply]
You’re right. I just tried it, and after draining the node with the attached volume, the CSI automatically attached it to the remaining node.

![CleanShot 2024-10-12 at 11 31 01@2x](https://github.com/user-attachments/assets/7620ddde-43b7-410a-a8e6-0b6d7fdd6ca5)

<img width="924" alt="Screenshot 2024-10-12 at 11 28 02" src="https://github.com/user-attachments/assets/3d6edd79-c565-49f1-aee8-4205d2b329c9">
